controller/stack_resources_test.go

```diff
@@ -1413,3 +1413,152 @@ func TestReconcileStackSecretRefs(t *testing.T) {
+	testPCSStack.Spec = zv1.StackSpecInternal{
+		StackSpec: zv1.StackSpec{
+			ConfigurationResources: []zv1.ConfigurationResourcesSpec{
+				{
+					PlatformCredentialsSet: &zv1.PCS{
+						Name: "foo-v1-test-pcs",
+					},
+				},
+			},
+		},
+	}
+
+	testPCS := zv1.PlatformCredentialsSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "foo-v1-test-pcs",
+			Namespace:       baseTestStackOwned.Namespace,
+			OwnerReferences: baseTestStackOwned.OwnerReferences,
+		},
+		Spec: zv1.PlatformCredentialsSpec{
+			Tokens: map[string]zv1.Token{
+				"token01": {
+					Privileges: []string{
+						"read-write",
+					},
+				},
+			},
+		},
+	}
+
+	updatePCSStack := baseTestStack
+	updatePCSStack.Spec = zv1.StackSpecInternal{
+		StackSpec: zv1.StackSpec{
+			ConfigurationResources: []zv1.ConfigurationResourcesSpec{
+				{
+					PlatformCredentialsSet: &zv1.PCS{
+						Name: "foo-v1-exist-pcs",
+					},
+				},
+			},
+		},
+	}
+
+	existingPCS := zv1.PlatformCredentialsSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "foo-v1-exist-pcs",
+			Namespace:       baseTestStackOwned.Namespace,
+			OwnerReferences: baseTestStackOwned.OwnerReferences,
+		},
+		Spec: zv1.PlatformCredentialsSpec{
+			Tokens: map[string]zv1.Token{
+				"token02": {
+					Privileges: []string{
+						"write",
+					},
+				},
+			},
+		},
+	}
+
+	updatedPCS := existingPCS
+	updatedPCS.Spec = zv1.PlatformCredentialsSpec{
+		Tokens: map[string]zv1.Token{
+			"token02": {
+				Privileges: []string{
+					"read",
+				},
+			},
+		},
+	}
+
+	for _, tc := range []struct {
+		name     string
+		stack    zv1.Stack
+		existing []*zv1.PlatformCredentialsSet
+		expected map[string]*zv1.PlatformCredentialsSet
+	}{
+		{
+			name:     "pcs is created if none exist",
+			stack:    testPCSStack,
+			existing: nil,
+			expected: map[string]*zv1.PlatformCredentialsSet{
+				"expected-pcs": &testPCS,
+			},
+		},
+		{
+			name:  "pcs is created when another already exists",
+			stack: testPCSStack,
+			existing: []*zv1.PlatformCredentialsSet{
+				&existingPCS,
+			},
+			expected: map[string]*zv1.PlatformCredentialsSet{
+				"expected-pcs": &testPCS,
+			},
+		},
+		{
+			name:  "pcs is updated if the stack changes",
+			stack: updatePCSStack,
+			existing: []*zv1.PlatformCredentialsSet{
+				&existingPCS,
+			},
+			expected: map[string]*zv1.PlatformCredentialsSet{
+				"expected-pcs": &updatedPCS,
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			env := NewTestEnvironment()
+
+			err := env.CreateStacksets(context.Background(), []zv1.StackSet{testStackSet})
+			require.NoError(t, err)
+
+			err = env.CreateStacks(context.Background(), []zv1.Stack{tc.stack})
+			require.NoError(t, err)
+
+			// Create case's existing resources
+			if tc.existing != nil {
+				for _, existing := range tc.existing {
+					err = env.CreatePCS(
+						context.Background(),
+						[]zv1.PlatformCredentialsSet{*existing},
+					)
+					require.NoError(t, err)
+				}
+			}
```

Comment:
Given that `CreatePCS` already takes a slice of PCS, we may be able to simplify this whole thing to:

```suggestion
			// Create case's existing resources
			err = env.CreatePCS(
				context.Background(),
				tc.existing,
			)
			require.NoError(t, err)
```

We might need to make `CreatePCS` to receive a list of pointers but I think that's a fine compromise.